templates/vite/src/lib.tsx

```diff
@@ -10,6 +10,7 @@ export const isTestnet = environment === environments.TESTNET;
 export const localProviderUrl = `http://127.0.0.1:${process.env.VITE_FUEL_NODE_PORT || 4000}/v1/graphql`;
 export const testnetProviderUrl = "https://testnet.fuel.network/v1/graphql";
 export const providerUrl = isLocal ? localProviderUrl : testnetProviderUrl;
+export const chainId = 0; // Local and testnet are both on chain 0
```

Comment:
Ideally this wouldn't be hardcoded as this template can be pointed at mainnet really easily right now (just by changing config). Is there anyway this can be fetched? Not a deal breaker but preferred.

Reply:
Not without an async request, which [doesn't seem](https://github.com/FuelLabs/fuels-ts/pull/3511/commits/f737f0e2768a6693369e910745b397c778650d18) to work.

Reply:
Could this be a thing on `@fuels/react:FuelProvider`?

Reply:
I might have misunderstood the application of this argument and have since [rolled back](https://github.com/FuelLabs/fuels-ts/pull/3511/commits/2ce732fe19a71076019956a6abb4f5832ea8c785).